protocol/src/lib.rs

```diff
@@ -0,0 +1,126 @@
+/// The default protocol used by the signalling server
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(tag = "peer-type")]
+#[serde(rename_all = "lowercase")]
+/// Confirms registration
+pub enum RegisteredMessage {
+    /// Registered as a producer
+    Producer,
+    /// Registered as a consumer
+    Consumer,
+    /// Registered as a listener
+    Listener,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(tag = "type")]
+#[serde(rename_all = "lowercase")]
+/// Messages sent from the server to peers
+pub enum OutgoingMessage {
+    /// Confirms registration
+    Registered(RegisteredMessage),
+    /// Notifies listeners that a producer was registered
+    ProducerAdded { peer_id: String },
+    /// Notifies listeners that a producer was removed
+    ProducerRemoved { peer_id: String },
+    /// Instructs a peer to generate an offer
+    StartSession { peer_id: String },
+    /// Signals that the session the peer was in was ended
+    EndSession { peer_id: String },
+    /// Messages directly forwarded from one peer to another
+    Peer(PeerMessage),
+    /// Provides the current list of consumer peers
+    List { producers: Vec<String> },
```

Comment:
```suggestion
    List { #[serde(default)] producers: Vec<String> },
```

Maybe

Reply:
hrm, what's the point of that? This is a message we emit fwiw

Reply:
Allowing to deserialize a message that doesn't contain the producers field at all

Reply:
hrm, the server will always provide it :)